cartographer/mapping/internal/3d/scan_matching/intensity_cost_function_3d.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_MAPPING_INTERNAL_3D_SCAN_MATCHING_INTENSITY_COST_FUNCTION_3D_H_
+#define CARTOGRAPHER_MAPPING_INTERNAL_3D_SCAN_MATCHING_INTENSITY_COST_FUNCTION_3D_H_
+
+#include "Eigen/Core"
+#include "cartographer/mapping/3d/hybrid_grid.h"
+#include "cartographer/mapping/internal/3d/scan_matching/interpolated_grid.h"
+#include "cartographer/sensor/point_cloud.h"
+#include "cartographer/transform/rigid_transform.h"
+#include "cartographer/transform/transform.h"
+#include "ceres/ceres.h"
+
+namespace cartographer {
+namespace mapping {
+namespace scan_matching {
+
+class IntensityCostFunction3D {
```

Comment:
Do you want to add a small summary as we have in the other cost functions? E.g. https://github.com/cartographer-project/cartographer/blob/ee98a92845f7ded3c7e509d32f245f3860d23001/cartographer/mapping/internal/3d/scan_matching/occupied_space_cost_function_3d.h#L31

Reply:
Done.